ssl/statem/extensions.c

```diff
@@ -1242,6 +1242,13 @@ int tls_psk_do_binder(SSL *s, const EVP_MD *md, const unsigned char *msgstart,
     const char *label;
     size_t bindersize, labelsize, hashsize = EVP_MD_size(md);
     int ret = -1;
+    int usepskfored = 0;
+
+    if (external
+            && s->early_data_state == SSL_EARLY_DATA_CONNECTING
+            && s->session->ext.max_early_data == 0
```

Comment:
It is theoretically possible for a client to provide *both* a ticket based session object *and* an external PSK based session object. In that case, if we are attempting to send early_data, then we prefer the ticket based session over the PSK. This check is about determining which one we are using. If `s->session->ext.max_early_data == 0` then the ticket based session does not support early_data.